src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -69,6 +70,21 @@ public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContex
                 return;
             }
 
+            // Need to special case for highlighting of method types because they are also "contained" within a method,
+            // but it does not make sense to introduce a parameter in that case. If the direct parent of the expression
+            // is a method, then it should not be able to have a parameter introduced for it.
+            if (expression.Parent is null)
+            {
+                return;
+            }
+
+            var methodSyntax = GetMethodSyntax(expression);
+            var directParentMethodSymbol = semanticModel.GetDeclaredSymbol(methodSyntax, cancellationToken);
+            if (directParentMethodSymbol is IMethodSymbol)
```

Comment:
i think a more appropriate check is to ask if you're in a Type's-Only context, not if your parent is a method.  Because what you're really asking is "do i have an expression that is a value?"

There are a couple of methods that can help out here. `IsInNamespaceOrTypeContext` for asking if this is only a type context (and thus not an expression).  And also, IsExpressionContext for checking that thsi is a valid location for a value to go.

Reply:
I def should've looked at what the syntaxfacts service had available, fixed.